pasta/base/annotate.py

```diff
@@ -132,7 +132,13 @@ def __init__(self):
 
   def visit(self, node):
     self._stack.append(node)
+    if hasattr(self, 'tokens'):
```

Comment:
I think you can move these to `AstAnnotor.visit` instead, it would remove the need for the `hasattr` check.

Reply:
Specifically around this line (1292)

`super(AstAnnotator, self).visit(node)`